File: java/ql/test/utils/modeleditor/com/github/codeql/test/PublicGenericClass.java
package com.github.codeql.test;

import java.nio.file.Paths;

public class PublicGenericClass<T, T2> implements PublicGenericInterface<T> {
  public void stuff(T arg) {
    System.out.println(arg);
  }

  public <T3> void stuff2(T3 arg) {
    System.out.println(arg);
  }
}

File: csharp/ql/test/utils/modeleditor/PublicGenericClass.cs
using System;

namespace GitHub.CodeQL;

public class PublicGenericClass<T, T2> : PublicGenericInterface<T>
{
    public void stuff(T arg)
    {
        Console.WriteLine(arg);
    }

    public void stuff2<T2>(T2 arg)
    {
        Console.WriteLine(arg);
    }

    public TNode summaryStuff<TNode>(TNode arg)
    {
        return arg;
    }
}
